src/Shabalov/Task-4/Widget.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include "Object3D.h"
+#include <QBasicTimer>
+#include <QMatrix4x4>
+#include <QMouseEvent>
+#include <QOpenGLBuffer>
+#include <QOpenGLFunctions>
+#include <QOpenGLShaderProgram>
+#include <QOpenGLTexture>
+#include <QOpenGLWidget>
+#include <QTimerEvent>
+#include <QWheelEvent>
+#include <memory>
+
+class Widget : public QOpenGLWidget, protected QOpenGLFunctions {
+  Q_OBJECT
+
+public:
+  explicit Widget(QWidget *parent = nullptr);
+  ~Widget() override = default();
```

Comment:
Этот код не скомпилируется
```suggestion
  ~Widget() override = default;
```